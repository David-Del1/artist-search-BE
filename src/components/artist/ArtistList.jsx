import React, { useState } from 'react';
import { useArtists } from '../../state/artist.js';
import Artist from './Artist';

const ArtistList = () => {

  const [searchedArtist, setArtist] = useState('artist\'s name');
  const [page, setPage] = useState(1);

  const { artists, loading } = useArtists(searchedArtist, page);

  if(loading) return <h1>Loading...</h1>;

  const artistElements = artists.map(artist => (
    <li key={artist.id}>
      <Artist {...artist} />
    </li>
  ));

  const handleSubmit = async (e) => {
    e.preventDefault();
    // const artists = await fetchArtists(searchedArtist, page);
    // return setArtist(artists);
  };

  return (
    <>
      <>
        <form onSubmit={handleSubmit}>
          <input placeholder="artists name" onChange={((e) => setArtist(e.target.value))} value={searchedArtist}></input>
        </form>
      </>
      <section>
        <ul>
          {artistElements}
        </ul>
        
        <button disabled={page <= 1} onClick={() => setPage((prevPage) => prevPage - 1)}>&lt;</button>
        <button disabled={artists.length < 10} onClick={() => setPage((prevPage) => prevPage + 1)}>&gt;</button>

      </section>
    </>
  );
};

export default ArtistList;

