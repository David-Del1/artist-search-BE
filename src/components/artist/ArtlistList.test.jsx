import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import ArtistList from './ArtistList';

describe('Artist List Component', () => {
  it('form, ul and two buttons', () => {

    const artist = [{
      'id': '1',
      'name': 'sam',
    }];

    const { asFragment } = render (
      <MemoryRouter>
        <ArtistList {...artist} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
