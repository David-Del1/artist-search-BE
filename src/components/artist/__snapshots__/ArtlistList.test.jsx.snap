// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist List Component form, ul and two buttons 1`] = `
<DocumentFragment>
  <h1>
    Loading...
  </h1>
</DocumentFragment>
`;
